apps/src/music/views/SoundsPanel2.tsx

```diff
@@ -0,0 +1,487 @@
+  );
+
+  return (
+    <div
+      className={classNames(
+        'sounds-panel-folder-row',
+        classNames(styles.folderRow, isSelected && styles.folderRowSelected)
+      )}
+      onClick={() => onSelect(folder)}
+      onKeyDown={event => {
+        if (event.key === 'Enter') {
+          onSelect(folder);
+        }
+      }}
+      ref={isSelected ? currentFolderRefCallback : null}
+      aria-label={folder.name}
+      tabIndex={0}
+      role="button"
+    >
+      <div className={styles.folderRowLeft}>
+        {imageSrc && (
+          <img src={imageSrc} className={styles.folderImage} alt="" />
+        )}
+      </div>
+      <div className={styles.folderRowMiddle}>
+        <div className={styles.folderRowMiddleName}>{folder.name}</div>
+        {folder.artist && (
+          <div className={styles.folderRowMiddleSubTitle}>{folder.artist}</div>
+        )}
+      </div>
+      <div className={styles.folderRowRight}>
+        <div className={styles.length}>&nbsp;</div>
+        {previewSound && (
+          <div className={styles.previewContainer}>
+            <FontAwesome
+              title={undefined}
+              icon={'play-circle'}
+              className={classNames(
+                styles.preview,
+                isPlayingPreview && styles.previewPlaying
+              )}
+              onClick={onPreviewClick}
+            />
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
+
+// SoundPanelEntry: A single sound.
+
+interface SoundsPanelEntryProps {
+  currentValue: string;
+  playingPreview: string;
+  folder: SoundFolder;
+  sound: SoundData;
+  showingSoundsOnly: boolean;
+  onSelect: (path: string) => void;
+  onPreview: (path: string) => void;
+  currentSoundRefCallback?: (ref: HTMLDivElement) => void;
+}
+
+const SoundsPanelEntry: React.FunctionComponent<SoundsPanelEntryProps> = ({
+  currentValue,
+  playingPreview,
+  folder,
+  sound,
+  onSelect,
+  onPreview,
+  currentSoundRefCallback,
+}) => {
+  const soundPath = folder.id + '/' + sound.src;
+  const isSelected = soundPath === currentValue;
+  const isPlayingPreview = playingPreview === soundPath;
+
+  const onSoundSelect = useCallback(() => {
+    if (!isPlayingPreview) {
+      onPreview(soundPath);
+    }
+    onSelect(soundPath);
+  }, [isPlayingPreview, onPreview, onSelect, soundPath]);
+
+  const onSoundClick = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      e.stopPropagation();
+      onSoundSelect();
+    },
+    [onSoundSelect]
+  );
+
+  return (
+    <div
+      className={classNames(
+        'sounds-panel-sound-entry',
+        styles.soundEntry,
+        isSelected && styles.soundEntrySelected
+      )}
+      onClick={onSoundClick}
+      onKeyDown={event => {
+        if (event.key === 'Enter') {
+          onSoundSelect();
+        }
+      }}
+      ref={isSelected ? currentSoundRefCallback : null}
+      aria-label={sound.name}
+      tabIndex={0}
+      role="button"
```

Comment:
Can this be a `<button>` component for better tab navigation?

Reply:
The keyboard navigation has been working well for a while in the previous version of this panel, so we might consider this for a follow-up when we next focus on overall navigation.